src/hooks/useL1CrossDomainMessengerProxy.ts

```diff
@@ -36,28 +35,13 @@ function useL2ContractCallInfo({
     chain,
   })
 
-  const { data: gas } = useSuspenseQuery({
-    queryKey: [
-      'estimateGas',
-      chain.id,
-      contract.address,
-      functionName,
-      args,
-      walletAddress,
-      value?.toString(),
-    ],
-    queryFn: () => {
-      const contract = getContract(contractName, chain.id)
-
-      return readOnlyClient.estimateContractGas({
-        address: contract.address,
-        abi: contract.abi,
-        functionName,
-        args: args as any, // TODO: TS does not infer correctly the type of value
-        account: walletAddress,
-        value: value as any, // TODO: TS does not infer correctly the type of value
-      })
-    },
+  const gas = await readOnlyClient.estimateContractGas({
+    address: contract.address,
+    abi: contract.abi,
+    functionName,
+    args: args as any, // TODO: TS does not infer correctly the type of value
+    account: walletAddress,
+    value: value as any, // TODO: TS does not infer correctly the type of value
```

Comment:
I checked the TS issue locally, by passing the generics to the `estimateContractGas` function TS stops complaining about `ags` type.

But it insist that `value` must be `undefined` tho.

```ts
  const gas = await readOnlyClient.estimateContractGas<
    typeof chain,
    typeof contract.abi,
    typeof functionName,
    typeof args
  >({
    abi: contract.abi,
    account: walletAddress,
    address: contract.address,
    functionName,
    args,
  })
```